app/server/lib/DocApi.ts

```diff
@@ -1199,22 +1210,48 @@ export class DocWorkerApi {
       } else {
         parameters = req.body;
       }
-      if (parameters.workspaceId) { throw new Error('workspaceId not supported'); }
+
+      const documentName = optStringParam(parameters.documentName);
+      const workspaceId = optIntegerParam(parameters.workspaceId);
       const browserSettings: BrowserSettings = {};
       if (parameters.timezone) { browserSettings.timezone = parameters.timezone; }
       browserSettings.locale = localeFromRequest(req);
+
+      let docId: string;
       if (uploadId !== undefined) {
-        const result = await this._docManager.importDocToWorkspace(userId, uploadId, null,
-                                                                   browserSettings);
-        return res.json(result.id);
+        ({id: docId} = await this._docManager.importDocToWorkspace({
+          userId,
+          uploadId,
+          documentName,
+          workspaceId,
+          browserSettings,
+        }));
+      } else if (workspaceId !== undefined) {
```

Comment:
Is this path covered in tests?

Reply:
Also passing workspaceId = 0 in the `form` flavor works (it ignores it), but here it doesn't. Maybe it is good to unify it.

Reply:
> Is this path covered in tests?

Yeah, the `without content` version of the new test should hit it.